File: 3.0/golang/2.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func main() {
	var k int
	var S string

	scanner := bufio.NewScanner(os.Stdin)
	const maxCapacity = 1000 * 1024
	buf := make([]byte, maxCapacity)
	scanner.Buffer(buf, maxCapacity)

	if scanner.Scan() {
		k, _ = strconv.Atoi(scanner.Text())
	}
	if scanner.Scan() {
		S = scanner.Text()
	}

	lns := len(S)
	alp := []rune{'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z'}

	ans := 0
	for _, letter := range alp {
		right := 0
		t := k
		count := 0
		for left := 0; left != lns-k; left++ {
			for right < lns && t >= 0 {
				if S[right] != byte(letter) {
					t--
				}
				right++
				count++
			}

			if t < 0 {
				count--
			}

			if S[left] != byte(letter) {
				t++
			}

			if count > ans {
				ans = count
			}
		}
	}

	fmt.Println(ans)
}

File: 3.0/python/2.py
k = int(input())
S = input()
lns = len(S)
alp = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z']
ans = 0

for letter in alp:
    right = 0
    t = k
    for left in range(lns - k):
        while right < lns and t >= 0:
            if S[right] != letter:
                t -= 1
            right += 1

        tmp = right - left - 1

        if S[left] != letter:
            t += 1
            
        if tmp > ans:
            ans = tmp

print(ans)